meta-mel-support/recipes-core/plymouth/plymouth-mel/mel.script

```diff
@@ -0,0 +1,162 @@
+bg_image = Image("special://logo");
+
+screen_ratio = Window.GetHeight() / Window.GetWidth();
+bg_image_ratio = bg_image.GetHeight() / bg_image.GetWidth();
+
+if (screen_ratio < bg_image_ratio)
+  {  # Screen ratio is taller than image ratio, we will match the screen height
+     scale_factor =  Window.GetHeight() / bg_image.GetHeight();
+  }
+else
+  {  # Screen ratio is wider than image ratio, we will match the screen width
+     scale_factor =  Window.GetWidth() / bg_image.GetWidth();
+  }
+#scale_factor = Window.GetWidth() / bg_image.GetWidth();
+
+scaled_bg_image = bg_image.Scale(bg_image.GetWidth()  * scale_factor,
+                                 bg_image.GetHeight() * scale_factor);
```

Comment:
Right, I didn't consider that. 

Minor changes are needed: 
When if condition is true, screen ratio is wider than image ratio so comment on line# 7 needs to be updated. 
Similarly comment in else condition needs to be updated (line# 11, taller instead of wider). 
Comment on line# 14 may be removed. 

Rest looks great. Thanks!